packages/@o3r/rules-engine/schematics/ng-update/v11.6/use-register-action-handlers.spec.ts

```diff
@@ -0,0 +1,100 @@
+      import {AppComponent} from './app/app.component';
+      import {PopupActionHandler} from './services/popup-action-handler';
+
+      bootstrapApplication(AppComponent, appConfig)
+        .then((m) => {
+          runInInjectionContext(m.injector, () => {
+            inject(RulesEngineRunnerService);
+            ruleEngine.actionHandlers.add(inject(PopupActionHandler));
+          });
+        })
+        // eslint-disable-next-line no-console
+        .catch(err => console.error(err));
+    `);
+    initialTree.create(fileWithActionHandlerDelete, `
+      import {inject, runInInjectionContext} from '@angular/core';
+      import {RulesEngineRunnerService} from '@o3r/rules-engine';
+      import {appConfig} from './app/app.config';
+      import {AppComponent} from './app/app.component';
+      import {PopupActionHandler} from './services/popup-action-handler';
+
+      bootstrapApplication(AppComponent, appConfig)
+        .then((m) => {
+          runInInjectionContext(m.injector, () => {
+            inject(RulesEngineRunnerService);
+            ruleEngine.actionHandlers.delete(inject(PopupActionHandler));
+          });
+        })
+        // eslint-disable-next-line no-console
+        .catch(err => console.error(err));
+    `);
+    initialTree.create(multiplePresenceFile, `
+      import {inject, runInInjectionContext} from '@angular/core';
+      import {RulesEngineRunnerService} from '@o3r/rules-engine';
+      import {ConfigurationRulesEngineActionHandler} from '@o3r/configuration/rules-engine';
+      import {appConfig} from './app/app.config';
+      import {AppComponent} from './app/app.component';
+      import {PopupActionHandler} from './services/popup-action-handler';
+
+      bootstrapApplication(AppComponent, appConfig)
+        .then((m) => {
+          runInInjectionContext(m.injector, () => {
+            inject(RulesEngineRunnerService);
+            ruleEngine.actionHandlers.add(inject(PopupActionHandler));
+            ruleEngine.actionHandlers.add(inject(ConfigurationRulesEngineActionHandler));
+            ruleEngine.actionHandlers.delete(inject(PopupActionHandler));
+            ruleEngine.actionHandlers.delete(inject(ConfigurationRulesEngineActionHandler));
+          });
+        })
+        // eslint-disable-next-line no-console
+        .catch(err => console.error(err));
+    `);
+  });
+
+  it('should not touch files without references', async () => {
+    expect(initialTree.readText(untouchedFile)).not.toContain('actionHandlers.add');
+    expect(initialTree.readText(untouchedFile)).not.toContain('actionHandlers.delete');
+    tree = await lastValueFrom(callRule(useRegisterActionHandlers, initialTree, {} as SchematicContext));
+    expect(tree.readText(untouchedFile)).toBe(initialTree.readText(untouchedFile));
+  });
+
+  it('should change file with a reference to actionHandlers.add', async () => {
+    expect(initialTree.readText(fileWithActionHandlerAdd)).toContain('actionHandlers.add');
```

Comment:
Isn't it weird to do an `expect` on the initial tree? We are just testing the code of the `beforeEach`?

Reply:
I added it to ensure that it is well present in the initial tree 
because I did the `find & replace` after writing the test, so indeed  `actionHandlers.add` was never present